src/cryptography/hazmat/bindings/openssl/x509_vfy.py

```diff
@@ -29,9 +29,22 @@
 typedef ... Cryptography_STACK_OF_ASN1_OBJECT;
 
 typedef ... X509_STORE;
-typedef ... X509_STORE_CTX;
 typedef ... X509_VERIFY_PARAM;
 
+typedef struct x509_store_ctx_st {
+    X509_STORE *ctx;
+    int current_method;
+    X509 *cert;
+    Cryptography_STACK_OF_X509 *untrusted;
+    Cryptography_STACK_OF_X509_CRL *crls;
+    X509_VERIFY_PARAM *param;
+    void *other_ctx;
+    int (*verify)(struct x509_store_ctx_st *ctx);
```

Comment:
Ah yeah, that should fix it.

On Wed, May 6, 2015 at 5:37 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/bindings/openssl/x509_vfy.py
> <https://github.com/pyca/cryptography/pull/1909#discussion_r29804968>:
>
> >  typedef ... X509_VERIFY_PARAM;
> >
> > +typedef struct x509_store_ctx_st {
> > +    X509_STORE *ctx;
> > +    int current_method;
> > +    X509 *cert;
> > +    Cryptography_STACK_OF_X509 *untrusted;
> > +    Cryptography_STACK_OF_X509_CRL *crls;
> > +    X509_VERIFY_PARAM *param;
> > +    void *other_ctx;
> > +    int (*verify)(struct x509_store_ctx_st *ctx);
>
> You've got an opaque typedef of X509_STORE_CTX above this, but you might
> get away with a forward declaration?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1909/files#r29804968>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084